aggregate/billing_functions/utils.py

```diff
@@ -0,0 +1,742 @@
+        ('disk/pd-ssd/1', rate_gib_month_to_mib_msec(0.23)),
+        # https://cloud.google.com/compute/disks-image-pricing#localssdpricing
+        ('disk/local-ssd/preemptible/1', rate_gib_month_to_mib_msec(0.065)),
+        ('disk/local-ssd/nonpreemptible/1', rate_gib_month_to_mib_msec(0.108)),
+        # legacy local-ssd
+        ('disk/local-ssd/1', rate_gib_month_to_mib_msec(0.108)),
+        # https://cloud.google.com/vpc/network-pricing#:~:text=internal%20IP%20addresses.-,External%20IP%20address%20pricing,to%0Athe%20following%20table.,-If%20you%20reserve
+        ('ip-fee/1024/1', rate_instance_hour_to_fraction_msec(0.004, 1024)),
+        # custom Hail Batch service fee?
+        ('service-fee/1', rate_cpu_hour_to_mcpu_msec(0.01)),
+    ]
+    s = json.dumps(dict(rates))
+    print(s)
+    return s
+
+
+AUSTRALIA_SOUTHEAST_1_COST = {
+    'compute/n1-preemptible/1': 2.4944444444444447e-12,
+    'compute/n1-nonpreemptible/1': 1.2466666666666668e-11,
+    'memory/n1-preemptible/1': 3.255208333333333e-13,
+    'memory/n1-nonpreemptible/1': 1.6303168402777778e-12,
+    'boot-disk/pd-ssd/1': 8.540929918624991e-14,
+    'disk/pd-ssd/1': 8.540929918624991e-14,
+    'disk/local-ssd/preemptible/1': 2.4137410639592365e-14,
+    'disk/local-ssd/nonpreemptible/1': 4.010523613963039e-14,
+    'disk/local-ssd/1': 4.010523613963039e-14,
+    'ip-fee/1024/1': 1.0850694444444444e-12,
+    'service-fee/1': 2.777777777777778e-12,
+}
+
+
+def get_usd_cost_for_resource(batch_resource, usage, region='australia-southeast-1'):
+    """
+    Get the cost of a resource in USD.
+    """
+    regions = {
+        'australia-southeast-1': AUSTRALIA_SOUTHEAST_1_COST,
+    }
+
+    return regions[region][batch_resource] * usage
+
+
+def get_unit_for_batch_resource_type(batch_resource_type: str) -> str:
+    """
+    Get the relevant unit for some hail batch resource type
+    """
+    return {
+        'boot-disk/pd-ssd/1': 'mib * msec',
+        'disk/local-ssd/preemptible/1': 'mib * msec',
+        'disk/local-ssd/nonpreemptible/1': 'mib * msec',
+        'disk/local-ssd/1': 'mib * msec',
+        'disk/pd-ssd/1': 'mb * msec',
+        'compute/n1-nonpreemptible/1': 'mcpu * msec',
+        'compute/n1-preemptible/1': 'mcpu * msec',
+        'ip-fee/1024/1': 'IP * msec',
+        'memory/n1-nonpreemptible/1': 'mib * msec',
+        'memory/n1-preemptible/1': 'mib * msec',
+        'service-fee/1': '$/msec',
+    }.get(batch_resource_type, batch_resource_type)
+
+
+def get_start_and_end_from_request(
+    request,
+) -> Tuple[Optional[datetime], Optional[datetime]]:
+    """
+    Get the start and end times from the cloud function request.
+    """
+    if request:
+        return request.params['start'], request.params['end']
+    return (None, None)
+
+
+def date_range_iterator(
+    start,
+    end,
+    intv=DEFAULT_RANGE_INTERVAL,
+) -> Iterator[Tuple[datetime, datetime]]:
+    """
+    Iterate over a range of dates.
+    """
```

Comment:
@violetbrina Add doctests to this function to test this function works as expected?

Reply:
Added some tests but not sure how to run them

Reply:
Adjusted the doctests to make them parse, I ran manually with pytest:

```shell
pytest --doctest-modules
```